rust/chains/abacus-ethereum/src/provider.rs

```diff
@@ -110,3 +106,25 @@ impl MakeableWithProvider for AbacusProviderBuilder {
         })
     }
 }
+
+/// Call a get function that returns a Result<Option<T>> and retry if the inner
+/// option is None. This can happen because the provider has not discovered the
+/// object we are looking for yet.
+async fn get_with_retry_on_none<T, F, O, E>(get: F) -> eyre::Result<T>
+where
+    F: Fn() -> O,
+    O: Future<Output = Result<Option<T>, E>>,
+    E: std::error::Error + Send + Sync + 'static,
+{
+    for i in 0..3 {
+        if i > 0 {
+            sleep(Duration::from_secs(5)).await;
```

Comment:
any reason to not put this before the `continue`? I figure if it fails once then it's unlikely to immediately work if retried again, so sleeping even on the first failure is beneficial - though maybe there's reason to believe that intuition is wrong

Reply:
still curious how you feel here. not passionate though

Reply:
Yeah, that makes sense, changing